### PR TITLE
Support for json schema

### PR DESCRIPTION
**Pull request by rlazo**

### CODE REVIEW

File: common/src/main/kotlin/com/google/ai/client/generativeai/common/APIController.kt [hunk 1]
```diff
@@ -240,7 +240,7 @@ private suspend fun validateResponse(response: HttpResponse) {
   if (message.contains("quota")) {
     throw QuotaExceededException(message)
   }
-  if (error.details.any { "SERVICE_DISABLED" == it.reason }) {
+  if (error.details?.any { "SERVICE_DISABLED" == it.reason } == true) {
```
Comment (rlazo) [inline common/src/main/kotlin/com/google/ai/client/generativeai/common/APIController.kt:243]: Why the extra == true?

File: generativeai-android-sample/app/build.gradle.kts [hunk 2]
```diff
@@ -75,5 +75,6 @@ dependencies {
     debugImplementation("androidx.compose.ui:ui-tooling")
```
Comment (rlazo) [inline generativeai-android-sample/app/build.gradle.kts]: no need to include this file just yet

File: generativeai/src/main/java/com/google/ai/client/generativeai/internal/util/conversions.kt [hunk 3]
```diff
@@ -99,7 +99,7 @@ internal fun com.google.ai.client.generativeai.type.GenerationConfig.toInternal(
     maxOutputTokens = maxOutputTokens,
     stopSequences = stopSequences,
     responseMimeType = responseMimeType,
-    schema = schema?.toInternal(),
+    responseSchema = schema?.toInternal(),
```
Comment (rlazo) [inline generativeai/src/main/java/com/google/ai/client/generativeai/internal/util/conversions.kt]: should this one be `responseSchema = responseSchema` ?

File: common/src/main/kotlin/com/google/ai/client/generativeai/common/APIController.kt (hunk 1 above)
Comment (davidmotson) [reply common/src/main/kotlin/com/google/ai/client/generativeai/common/APIController.kt:243]: required because of the possibility of a null

Comment (rlazo) [reply common/src/main/kotlin/com/google/ai/client/generativeai/common/APIController.kt:243]: it should be 

```
error.details?.any { "SERVICE_DISABLED" == it.reason } ?: false
```

then

**Review by davidmotson — APPROVED** [review]

**Pull request merged**